src/components/react-json-schemaForm/GeometryField/GeometryField.js

```diff
@@ -1,48 +1,219 @@
-import React from 'react';
+import React, { useState, useCallback, useEffect, useMemo } from 'react';
+import { CONTROL_DRAW, CONTROLS_TOP_LEFT } from '@terralego/core/modules/Map';
 import PropTypes from 'prop-types';
-import Message from '../../Message';
+import { ACTION_CREATE, getView } from '../../../modules/CRUD/services/CRUD';
+
+import {
+  ALL,
+  POINT,
+  LINESTRING,
+  POLYGON,
+  MULTI_POINT,
+  MULTI_LINESTRING,
+  MULTI_POLYGON,
+} from '../../../utils/geom';
+
 import PointField from './PointField';
+import DefaultField from './DefaultField';
 
-const Fieldset = ({ t, children }) => (
-  <fieldset>
-    <legend>{t('jsonSchema.geometryField.title')}</legend>
-    <div className="form-group field">
-      {children}
-    </div>
-  </fieldset>
-);
+const getLayerPaintId = (layerPaints, name) => {
+  const { id } = layerPaints.find(({ 'source-layer': source }) => source === name) || {};
+  return id;
+};
 
-const GeometryField = props => {
-  const { formData: { coordinates, type }, t } = props;
-  if (type === 'Point') {
-    return (
-      <Fieldset t={t}>
-        <PointField {...props} />
-      </Fieldset>
-    );
+const resetStyle = (map, layerPaints, name) => {
+  if (!Object.keys(map).length) {
+    return;
+  }
+  const layerPaintId = getLayerPaintId(layerPaints, name);
+  map.setFilter(layerPaintId, null);
+};
+
+/**
+ * Get geometries from
+ * -settings informations if this is a new one
+ * -feature's stored if this is an update
+ *
+ * @param {Object} {
+ *   feature,
+ *   formData: { coordinates, type },
+ *   params: { id, layer },
+ *   name,
+ *   settings,
+ * }
+ * @returns an object wrapping
+ * the name, identifier, geometry and endpoint (to save the feature) to manipulate
+ */
+const getGeometries = ({
+  feature,
+  formData: { coordinates, type },
+  params: { id, layer },
+  name,
+  settings,
+}) => {
+  if (id === ACTION_CREATE) {
+    const { layer: { name: layerName, ...rest }, featureEndpoint } = getView(settings, layer);
+    return {
+      ...rest,
+      identifier: null,
+      geom: { coordinates, type },
+      layerName,
+      url: featureEndpoint,
+    };
   }
-  const action = !coordinates.length ? 'create' : 'edit';
+  const { geometries: { [name]: featureName } } = feature[id];
+  return { layerName: name, ...featureName };
+};
+
+
+const GeometryField = props => {
+  const {
+    addControl,
+    feature,
+    formData,
+    layerPaints,
+    map,
+    match: { params },
+    removeControl,
+    name,
+    settings,
+    t,
+    ...rest
+  } = props;
+
+  const geometries = useMemo(() => (
+    getGeometries(({
+      feature,
+      formData,
+      params,
+      name,
+      settings,
+    }))
+  ), [feature, formData, params, name, settings]);
```

Comment:
Inlining object properties could improve visual similarity with hook dependencies.

```suggestion
  const geometries = useMemo(() => (
    getGeometries(({ feature, formData, params, name, settings }))
  ), [feature, formData, params, name, settings]);
```